sky/backends/cloud_vm_ray_backend.py

```diff
@@ -208,6 +208,9 @@ def add_prologue(self,
             self._code += [
                 textwrap.dedent(f"""\
                 _SETUP_CPUS = 0.0001
+                # Unset CUDA_VISIBLE_DEVICES (set by ray), since the setup command
+                # may need to detect the number of GPUs on the machine.
+                setup_cmd = 'unset CUDA_VISIBLE_DEVICES; ' + setup_cmd
```

Comment:
What happens if user does something like
```
setup: |
  export CUDA_VISIBLE_DEVICES=1
  python setup_gpu.py
```
and run `sky launch --detach-setup`?

It seems like this will unset user's intent.

Reply:
It should not, since the `unset` will happens before the setup script is run, so any command inside the setup script will override it.

Reply:
Added the setting for setting the `CUDA_VISIBLE_DEVICES` to the test.

Tested:
- [ ] `sky launch -c env-check ./examples/env_check.yaml`